### DISCUSSION TITLE
missing vesion label produces a confusing warning message in the UI

### DISCUSSION THREAD

**Discussion by jmazzitelli** [kiali/kiali#5321] [Q&A]
There is a message in the UI that is confusing. A customer asked about it, and I didn't have an answer.

I would like to know if (a) we should remove this message or (b) if it is really valid, we should explain it further (either have more details in the message itself or an FAQ).

See: https://github.com/kiali/kiali/blob/v1.53.0/frontend/src/components/MissingLabel/MissingLabel.tsx#L43

```javascript
        {this.props.missingVersion && (
          <>
            <div>
              <PFBadge badge={{ badge: versionLabel }} isRead={true} style={{ marginRight: '0px' }} /> label is missing.{' '}
              <br />
            </div>
            <div>This workload won't have Istio routing capabilities.</div>`
```

So for this pod, there is no version label. The message `This workload won't have Istio routing capabilities.` seems incorrect. As Rob Cernich stated, it's a bit of a misstatement because routing can use any label. It's just that "version" is typically used for things like a/b, blue/green, etc., but subsetting could also be done based on other labels, so this is really a convention/best practice kind of thing as opposed to something that might not work. In other words, routing is managed through subsets which can use whatever label selectors they want.

If this is true, I think we should either remove that message or re-word this - perhaps say, `This workload won't have Istio routing capabilities for routing configurations using the version label`. Or something like that. Or, as I say, maybe include an FAQ (or add an FAQ in addition to re-wording??).

I don't know the solution - but on the surface, it appears this message is incorrect and causing confusing. So that confusion needs to be addressed.

**Comment by israel-hdez** [comment]
My only idea is that since Kiali relies quite a lot in those labels, probably the _management_ features may depend on them (for example, the wizards). So, if workloads/services don't have those labels, it may not be possible to configure routing using Kiali, or it may misbehave.

That said, I'm not very familiar with that area, so I may be wrong. What is true is that, as you say, those labels are simply an Istio recommendation and routing can be configured using other labels.
